airflow/models/expandinput.py

```diff
@@ -57,9 +57,9 @@ class MappedArgument(ResolveMixin):
     _key: str
 
     def get_task_map_length(self, run_id: str, *, session: Session) -> int | None:
-        # TODO (AIP-42): Implement run-time task map length inspection.
-        # This simply marks the value as un-expandable at parse-time.
-        return None
+        # TODO (AIP-42): Implement run-time task map length inspection. This is
+        # needed when we implement task mapping inside a mapped task group.
+        raise NotImplementedError()
```

Comment:
```suggestion
        # TODO (AIP-42): Implement run-time task map length inspection. This is
        # needed when we implement task mapping inside a mapped task group.
        raise NotImplementedError()
```

is the todo still needed? or perhaps does the TODO need qualifying re where we actually do need to implement?

Reply:
_Task mapping inside a mapped task group_ as in

```python
@task_group
def tg(a):
    @task
    def f(x):
        print(x)

    # This is currently not allowed to simplify UI and API.
    f.expand(x=a)

tg.expand(a=[[1, 2], [3, 4]])
```